src/main/java/com/dropbox/core/v2/callbacks/DbxGlobalCallbackFactory.java

```diff
@@ -0,0 +1,17 @@
+package com.dropbox.core.v2.callbacks;
+
+/**
+ * Client should implement methods in this class to allow for consistent, global handling of route-specific error types,
+ * as well as general network errors. Normally, error handling is done on a request-by-request basis.
+ * However, it is convenient to handle some error behavior consistently, regardless of the request or the endpoint.
+ * For implementing global error handling for general networking errors like an auth error, `createNetworkErrorCallback`
+ * should be implemented. For implementing global error handling for route-specific errors, the
+ * `createRouteErrorCallback` should be implemented.
+ */
+public interface DbxGlobalCallbackFactory {
+    // Should instantiate separate callback object on each method invocation
+    <T> DbxRouteErrorCallback<T> createRouteErrorCallback(T routeError);
```

Comment:
I think you can overload these, i.e. have methods like 

```
DbxRouteErrorCallback<ListFolderError> createRouteErrorCallback(ListFolderError routeError);
DbxRouteErrorCallback<LookupError> createRouteErrorCallback(LookupError routeError);
...
```
It could also be an abstract class so implementors need only add the ones they care about.

Just suggesting an alternative that doesn't use a bunch of if/elses (if it even works? my Java is not that great), but feel free to keep it this way since this is fine too.